File: src/main/java/br/com/useinet/finance/dto/AuthResponse.java
package br.com.useinet.finance.dto;

public class AuthResponse {
    private String token;
    private String refreshToken;
    private String nome;
    private String email;

    public AuthResponse(String token, String refreshToken, String nome, String email) {
        this.token = token;
        this.refreshToken = refreshToken;
        this.nome = nome;
        this.email = email;
    }

    public String getToken() { return token; }
    public String getRefreshToken() { return refreshToken; }
    public String getNome() { return nome; }
    public String getEmail() { return email; }
}

File: frontend/lib/core/models/auth_response.dart
class AuthResponse {
  final String token;
  final String refreshToken;
  final String nome;
  final String email;

  AuthResponse({
    required this.token,
    required this.refreshToken,
    required this.nome,
    required this.email,
  });

  factory AuthResponse.fromJson(Map<String, dynamic> json) => AuthResponse(
        token: json['token'] as String,
        refreshToken: json['refreshToken'] as String? ?? '',
        nome: json['nome'] as String,
        email: json['email'] as String,
      );
}